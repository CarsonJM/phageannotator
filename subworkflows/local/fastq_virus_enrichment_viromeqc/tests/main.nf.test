nextflow_workflow {

    name "Test subworkflow: FASTQ_VIRUS_ENRICHMENT_VIROMEQC"
    script "subworkflows/local/fastq_virus_enrichment_viromeqc/main.nf"
    workflow "FASTQ_VIRUS_ENRICHMENT_VIROMEQC"
    tag "subworkflows"
    tag "subworkflows_local"
    tag "fastq_virus_enrichment_viromeqc"
    tag "fastq_virus_enrichment_viromeqc_default"


    test("Parameters: default") {

        when {
            workflow {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    [
                        file(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkifExists: true),
                        file(params.test_data['sarscov2']['illumina']['test_2_fastq_gz'], checkifExists: true),
                    ]
                ]
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() }
            )
        }
    }
}
