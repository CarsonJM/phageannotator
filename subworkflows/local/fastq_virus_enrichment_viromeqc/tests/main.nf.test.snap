{
    "Parameters: default": {
        "content": [
            {
                "0": [
                    "test.viromeqc.tsv:md5,6acbffc97f75774cf9fcc953b98913ed"
                ],
                "1": [
                    "versions.yml:md5,c26288b67351c3bdaad635c87a0249e7",
                    "versions.yml:md5,f771871e8cbe81944f6d5fbbc56cc7a5"
                ],
                "enrichment_tsv": [
                    "test.viromeqc.tsv:md5,6acbffc97f75774cf9fcc953b98913ed"
                ],
                "versions": [
                    "versions.yml:md5,c26288b67351c3bdaad635c87a0249e7",
                    "versions.yml:md5,f771871e8cbe81944f6d5fbbc56cc7a5"
                ]
            }
        ],
        "timestamp": "2023-12-28T13:57:38.935414881"
    }
}